ols/src/cache/redis_cache.py

```diff
@@ -22,29 +18,34 @@ class RedisCache(Cache):
     _instance = None
     _lock = threading.Lock()
 
-    def __new__(cls):
+    def __new__(cls: type["RedisCache"], config: RedisConfig) -> "RedisCache":
         """Create a new instance of the `RedisCache` class."""
         with cls._lock:
             if not cls._instance:
                 cls._instance = super(RedisCache, cls).__new__(cls)
-                cls._instance.initialize_redis()
+                cls._instance.initialize_redis(config)
         return cls._instance
 
-    def initialize_redis(self) -> None:
+    def initialize_redis(self, config: RedisConfig) -> None:
         """Initialize the Redis client and logger.
 
         This method sets up the Redis client with custom configuration parameters.
         """
+        kwargs = {}
```

Comment:
@tisnik The idea here is to pass username or password only if they are defined in the passed-in config context, and leave them as defaults defined in StrictRedis(). Is this saying what I want it to say?